### ISSUE TITLE
MENC_EVENT_SECURE  for srtp?

### ISSUE THREAD

**Issue by juha-h** [alfredh/baresip#657]
What would it take to issue MENC_EVENT_SECURE event when SRTP becomes enabled in srtp.c start_crypto() function?

Now only an info message is printed.  I guess access to session would be needed in order to call its event handler.

**Comment by juha-h** [comment]
Alterntively, is there some API means to figure out if call media is using SRTP, when account's mediaenc=srtp, i.e., when SRTP is optional?

**Comment by alfredh** [comment]
the same event has to be added to the srtp module, similar to
dtls_srtp and zrtp:

```c
	if (ds->sess->eventh) {
		if (re_snprintf(buf, sizeof(buf), "%s,%s",
				sdp_media_name(ds->sdpm),
				comp->is_rtp ? "RTP" : "RTCP"))
			(ds->sess->eventh)(MENC_EVENT_SECURE, buf,
					   ds->sess->arg);
		else
			warning("dtls_srtp: failed to print secure"
				" event arguments\n");
	}
```

also we could add a function in call/stream to check if the media stream
is secure:

```c
bool stream_is_encrypted(const struct stream *strm);
```

this one could return a flag that is set here:

```
static void menc_event_handler(const enum menc_event event,
			       const char *prm, void *arg)
{
	struct call *call = arg;
	MAGIC_CHECK(call);

	if (str_isset(prm))
		call_event_handler(call, CALL_EVENT_MENC, "%u,%s", event,
				   prm);
	else
		call_event_handler(call, CALL_EVENT_MENC, "%u", event);
}
```

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> the same event has to be added to the srtp module, similar to
> dtls_srtp and zrtp:

I was thinking about doing it, but then realized that srtp code does not
currently have
``
	const struct menc_sess *sess;
``
in struct menc_st.  Should one be added?

**Comment by juha-h** [comment]
juha-h writes:

> I was thinking about doing it, but then realized that srtp code does not
> currently have
> ``
> 	const struct menc_sess *sess;
> ``
> in struct menc_st.  Should one be added?

I guess it would also mean that a session_alloc function should be
written that can replace the NULL here:
``
static struct menc menc_srtp_opt = {
	LE_INIT, "srtp", "RTP/AVP", NULL, alloc
};
``
Looks like a rather big rewrite of the current code.

**Comment by juha-h** [comment]
Please check https://github.com/alfredh/baresip/pull/660.  It has worked OK in my tests.

**Comment by alfredh** [comment]
fixed in 489a40f82e907d85c99e8b26e45bedae5272e30f
